File: 이지수/Python/Baekjoon_13423.py
import sys
input = sys.stdin.readline

t = int(input().rstrip())
for _ in range(t) :
    n = int(input().rstrip())
    cnt = 0
    x = list(map(int, input().rstrip().split()))
    d = {i:True for i in x}
    for i in range(n):
        for j in range(i+1, n) :
            if (x[i]+x[j])/2 in d : cnt += 1
    print(cnt)

File: 이지수/Swift/Baekjoon_13423.swift
import Foundation
let t = Int(readLine()!)!
for _ in 0..<t {
    let n = Int(readLine()!)!
    let x = readLine()!.split(separator: " ").map(){Int($0)!}
    var d = [Double: Bool]()
    var cnt = 0
    for i in x { d[Double(i)] = true }
    for i in 0..<n {
        for j in i+1..<n where d[Double(x[i]+x[j])/Double(2)] != nil {
            cnt += 1
        }
    }
    print(cnt)
}
